src/user-mixins/poolify.ts

```diff
@@ -40,7 +40,7 @@ const poolifyMixin = <MixinBase extends Constructor>(mixinBase: MixinBase) => <T
       return this._pool!  // FIXME: why we need "!" at here?
     }
 
-    public static load<L extends Constructor<T> & PoolifyMixin<T>> (
+    public static load<L extends Constructor<T & {}> & PoolifyMixin<T>> (
```

Comment:
Why do we need to add `& {}` here?

Can you explain in detail?

And does it a workaround and need to be fixed in the future versions, or it should be kept with a reasonable explanation?

Reply:
This should be kept because Constructor type now reaquires the first parameter to extend {}.
```ts
 type Constructor<
  I extends {} = {},
  C = {}
> = ClassInterface<C> & InstanceInterface<I>
```

Reply:
I tried to add <T extends {}> here:
```ts
const poolifyMixin = <MixinBase extends Constructor>(mixinBase: MixinBase) => <T>() => 
```

But it somehow breaks the whole system.

Reply:
This is very tricky because with your modification, your code will allow to pass an `T` does not extends `{}` to be passed to the `Constructor`, which is most likely a workaround.

Reply:
Yeah the best practice I believe is to make T extend {} in the mixn declaration. But that will break other user-modules. (recursively references).
I don't know the reason for now.

Reply:
That's why I'm asking whether it's a workaround that should be fixed in the future or if it should be kept as a regular code.

So it looks more like a workaround now. Could you please file an issue to record this and add the issue link to the source code you modified? Because doing so, we can track the workaround and make the maintenance work in the future possible.

Thank you very much!

Reply:
https://github.com/wechaty/wechaty/issues/2553

Reply:
Could you please add comments in the code to brief explain and link to the issue?

thank you very much!

Reply:
Sure and done